### PR TITLE
Spiking out some ideas

### PR DESCRIPTION
**Pull request by bmulvihill**
For discussion

- Possibly using a partial for "exercise specific" template logic
- Add bookkeeping to test_cases
- Move template structure into the template

I had some trouble with clean indentation -but this is more of a spike branch than a final implementation so I just got it "working". I also didn't update any tests for the same reason.

### CODE REVIEW

File: lib/exercise_test_cases.rb [hunk 1]
```diff
@@ -15,6 +16,6 @@ def parsed_json_cases
   end
 
   def to_a
-    parsed_json_cases.map { |test_case| case_classname.new(test_case) }
+    parsed_json_cases.map.with_index { |test_case, index| case_classname.new(test_case, index) }
```
Comment (Insti) [inline lib/exercise_test_cases.rb]: `index` should not be an argument to the test case.

File: lib/generator.rb [hunk 2]
```diff
@@ -4,13 +4,14 @@
 require 'json'
 require 'ostruct'
 
-require_relative 'exercise_testcases'
-require_relative 'exercise_testcase'
+require_relative 'exercise_test_cases'
+require_relative 'exercise_test_case'
```
Comment (Insti) [inline lib/generator.rb:8]: one word? two words? I guess it's two..

Comment (bmulvihill) [reply lib/generator.rb:8]: I just changed it to follow the ruby class/file naming convention (the class was TestCase)

File: lib/exercise_test_cases.rb (hunk 1 above)
Comment (bmulvihill) [reply lib/exercise_test_cases.rb]: yeah I agree, it could be moved to be a parameter of the skip method?

Comment (Insti) [reply lib/exercise_test_cases.rb]: I guess it needs to go to skip now, that was one of the nice things about having a `render` method.

File: lib/generator.rb (hunk 2 above)
Comment (Insti) [reply lib/generator.rb:8]: :+1: on Ruby conventions

File: lib/exercise_test_cases.rb (hunk 1 above)
Comment (bmulvihill) [reply lib/exercise_test_cases.rb]: Yeah I was kind of torn about whether or not to keep the skip method - I think there is benefit to both ways for sure.

Comment (bmulvihill) [reply lib/exercise_test_cases.rb]: The main benefit for keeping it is so it is obvious that the skip method is part of the test structure.  That way if there is a test that needs some sort of custom skip logic (i.e. long running tests or some unforeseen issue) it is obvious where that would be implemented. 

The main benefit for encapsulating it in the render method is for template simplicity, and leaving less questions about what a new generator actually needs to implement.

Comment (Insti) [reply lib/exercise_test_cases.rb]: Could we make another file which is a partial for the template for a method?

```
<%= test_case.comment %>
  def <%= test_case.test_name %>
    <%= test_case.skip %>
    <%= test_case.render %>
  end
```

**Comment by bmulvihill** [comment]
@Insti I moved the index to this skip method, how do you think that looks?  I don't know if Im following the comment about a partial.

**Pull request closed without merging**